src/main/java/com/iota/iri/utils/log/interval/IntervalLogger.java

```diff
@@ -0,0 +1,490 @@
+     *
+     * It checks if the given message is new and then triggers the output.
+     *
+     * Error messages will always get dumped immediately instead of scheduling their output. Since error messages are
+     * usually a sign of some misbehaviour of the node we want to be able to see them when they appear (to be able to
+     * track down bugs more easily).
+     */
+    @Override
+    public IntervalLogger error(String message, Throwable cause) {
+        Message newMessage = new ErrorMessage(message, cause);
+
+        if (!newMessage.equals(lastReceivedMessage)) {
+            lastReceivedMessage = newMessage;
+
+            triggerOutput(true);
+        }
+
+        return this;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean getEnabled() {
+        return enabled;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public IntervalLogger setEnabled(boolean enabled) {
+        this.enabled = enabled;
+
+        return this;
+    }
+
+    /**
+     * Does the same as {@link #triggerOutput(boolean)} but defaults to letting the logger decide when to print the
+     * message.
+     */
+    public void triggerOutput() {
+        triggerOutput(false);
+    }
+
+    /**
+     * Triggers the output of the last received message.
+     *
+     * It either prints the message immediately (if enough time has passed since the last output or requested by the
+     * caller) or schedules it for the next interval.
+     *
+     * @param printImmediately flag indicating if the messages should be scheduled or printed immediately
+     */
+    public void triggerOutput(boolean printImmediately) {
+        if (lastReceivedMessage != null && (printImmediately ||
+                System.currentTimeMillis() - lastLogTime >= logInterval)) {
+
+            lastReceivedMessage.print();
+
+            if (outputThread != null) {
+                synchronized (this) {
+                    if (outputThread != null) {
+                        outputThread.interrupt();
+                    }
+                }
+            }
+        } else {
+            scheduleOutput();
+        }
+    }
+
+    /**
+     * This method schedules the output of the last received message by spawning a {@link Thread} that will print the
+     * new message after the given timeout.
+     *
+     * When creating the {@link Thread} it copies its name so the output is "transparent" to the user and the effect
+     * is the same as dumping the message manually through the {@link org.slf4j.Logger} object itself.
+     */
+    private void scheduleOutput() {
+        if (outputThread == null) {
+            synchronized(this) {
+                if (outputThread == null) {
+                    outputThread = new Thread(() -> {
+                        try {
+                            Thread.sleep(Math.max(logInterval - (System.currentTimeMillis() - lastLogTime), 1));
+
+                            outputThread = null;
```

Comment:
Recycling a thread? As far as i am aware you can not restart a thread. The only other way would be to have a thread running all the time and just repeating its task every 3 seconds which would be a massive overhead.

Imagine i call sth like

intervalLogger.info("Bla");

then i dont call the intervalLogger for like 30 minutes

and then i do another 

intervalLogger.info("Blub");

If we would want to use reuse the same output thread it would have to keep running doing unnecessary checks all the time in the background even tho for this scenario a thread wouldnt even get started

Reply:
Well executors recycle threads (look at the javadocs in `Executors`). It is usually a better practice to let those thread pools manage thread pools for you. See https://stackoverflow.com/questions/5483047/why-is-creating-a-thread-said-to-be-expensive.